Allow 13 asterisk characters in password field without warning

kucharskim opened the pull request:
##### SUMMARY

On OpenBSD, 13 asterisk characters as a password hash, marks the
account as disabled. Otherwise daily(8) script which executes
security(8) will email operator about not properly locked accounts.

Before the diff, we see following warning:

> [WARNING]: The input password appears not to have been hashed. The 'password' argument must be encrypted for this module to work properly.

After the diff, warning is gone.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
- modules/system/user.py

##### ADDITIONAL INFORMATION

I can provide more info, if needed.

kucharskim commented:
I tried to unit tests myself my diff, but so far didn't had luck to run tests, so my diff is tested live on OpenBSD systems, without execution of unit tests. Unit tests use a lot of hardcoded paths to python (which I needed to modify) and it seems framework is not compatible with non-wxallowed mount(8) points on OpenBSD.

samdoran reviewed (commented):
Change looks good. Thank you for adding tests. Please create a changelog fragment. See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

kucharskim commented:
@samdoran, please let me know does the changelog fragment looks good, or do I need to make some additional changes. Thanks.

kucharskim committed:
Allow 13 asterisk characters in password field without warning

On OpenBSD, 13 asterisk characters as a password hash, marks the
account as disabled. Otherwise daily(8) script which executes
security(8) will email operator about not properly locked accounts.

Before the diff, we see following warning:

> [WARNING]: The input password appears not to have been hashed. The 'password' argument must be encrypted for this module to work properly.

After the diff, warning is gone.

kucharskim commented:
@samdoran, I didn't hear back from anyone in recent weeks. Do you guys need me to do anything more here? Could this be merged?

samdoran commented:
rebuild_merge
